### PR TITLE
Microsoft.Data.Sqlite: Don't use bool? in connection string builder

### PR DESCRIPTION
**Pull request by bricelam**
Note, this is done in a way that won't break existing connection strings.

### CODE REVIEW

**Comment by bricelam** [comment]
I found a consumer that breaks because of this, and decided it was probably worth it after all.

**Review by smitpatel — APPROVED** [review]

**Comment by roji** [comment]
Out of pure curiosity, how was this breaking the consumer?

**Comment by bricelam** [comment]
lol, here's the totally awesome code that throws:
```cs
foreach (PropertyDescriptor property in TypeDescriptor.GetProperties(connectionStringBuilder))
{
    // NotSupportedException: DataSet does not support System.Nullable<>.
    dataTable.Columns.Add(property.Name, property.PropertyType);
}
```
It's in the [DDEX SDK](https://docs.microsoft.com/previous-versions/visualstudio/visual-studio-2013/bb165128(v=vs.120)). I've been having some fun over in [bricelam/VS.Data.Sqlite](https://github.com/bricelam/VS.Data.Sqlite). More PRs to come... 😉

File: src/Microsoft.Data.Sqlite.Core/SqliteForeignKeys.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+{
+    /// <summary>
+    ///     Represents the foreign key constraint enforcement modes of SQLite.
+    /// </summary>
+    public enum SqliteForeignKeys
+    {
+        /// <summary>
+        ///     The database default.
+        /// </summary>
+        Default = -1,
```
Comment (bricelam) [inline src/Microsoft.Data.Sqlite.Core/SqliteForeignKeys.cs:14]: I'd like to name this `NullableBoolean` with members `True`, `False`, and `Default`. But those are all VB keywrods... Let's discuss this whole PR in a design meeting.

**Comment by ErikEJ** [comment]
Wu hu - maybe EF Core Power Tools can finally get rid of System.Data.SQLite DDEX 🥳

**Comment by roji** [comment]
But... I *just* threw out the Npgsql DDEX VSIX for 5.0 - and you're making one for Sqlite? :rofl:

**Comment by bricelam** [comment]
@ErikEJ I definitely want to work with you to figure out the future of this thing. Right now, I'm just having fun, and trying to make our ADO.NET provider better. If nothing else, at least the world will have another sample showing how to implement a DDEX provider.

image

image

**Comment by bricelam** [comment]
Also, I finally understand the madness behind DbConnection.GetSchema(). IMHO, they put it in the wrong layer--it never should have been part of ADO.NET.
```xml
<Selection type="Column"
           restrictions="{Table.Catalog},{Table.Schema},{Table.Name}" /> <!-- restrictionValues -->
```
```xml
<Service type="IVsDataObjectSelector"
         implementationType="Microsoft.VisualStudio.Data.Framework.AdoDotNet.AdoDotNetObjectSelector">
  <Parameters method="SelectObjects">
    <Parameter value="Columns" /> <!-- collectionName -->
  </Parameters>
</Service>
```
```cs
connection.GetSchema("Columns", new[] { "Northwind", "dbo", "Categories" })
```

**Comment by ErikEJ** [comment]
@bricelam just so I understand - add connection only works with changes to the ADO.NET provider?

**Comment by roji** [comment]
Am interested to see where you land with this... I did invest quite a bit in the Npgsql DDEX VSIX at the time.

**Comment by bricelam** [comment]
@ErikEJ Yes, I’m using my [bricelam:ddex2](https://github.com/dotnet/efcore/compare/main...bricelam:ddex2) branch to develop locally. I’ll slowly merge the changes into 6.0 once I decide they’re worth it (and not worth just working around in the DDEX provider)

**Comment by bricelam** [comment]
Design Meeting notes: Still not worth it. I've since [worked around](https://github.com/bricelam/VS.Data.Sqlite/blob/be47fd967245785ccf77fb1e5b0334676d2c839a/VS.Data.Sqlite/SqliteRootObjectSelector.cs#L10) the DDEX code.

**Pull request closed without merging**